### DISCUSSION TITLE
How to suppress space added before initial stroke?

### DISCUSSION THREAD

**Discussion by triallax** [opensteno/plover#1533] [Q&A and Support] [answered]
I'm somewhat of a newbie to Plover and stenography in general (both of which I am really enjoying quite much), 
and I've went through most of [Art of Chording](https://www.artofchording.com/). However, one point that the book doesn't mention (or if it did, then I missed it) is how to handle Plover outputting a space before the first stroke. I've looked online and found that `TK-LS` suppresses the space before a stroke. Is that the "official" way to avoid an unneeded space at the beginning of a sentence?

Edit: I also found out about `KPA*`, is that what I'm supposed to use?

**Comment by JoshuaGrams** [comment, marked as answer]
`KPA*` will capitalize and suppress a space, yeah. `TK-LS` works if you want lowercase. You might have a look at the [Dictionary Format page](https://github.com/openstenoproject/plover/wiki/Dictionary-Format) for more info about defining formatting commands: there are a couple of common things (like `HRO*ER` for lowercase) that aren't in the default dictionary...

**Reply by triallax** [reply]
Alright, thanks for the answer!

**Comment by triallax** [comment]
So I just found out that you can disable the initial space and enable capitalization of the first word and  by going to the Output tab in the settings, and enabling "Start attached" and "Start capitalized" respectively.

**Reply by JoshuaGrams** [reply]
Yes, those are convenient, but they only affect the first word after launching Plover. So they don't help when you switch applications or switch form fields or whatever. You do still need to know about `KPA*` and `TK-LS` and `S-P`.

It's a common beginner (and even not-so-beginner) thing for people to want Plover to do a better job of knowing when you might want a space or capitalization. But Plover doesn't know which application you're in (though it *could* on most platforms, I think?). And even if it did, it doesn't know about how your text editor or word processor works exactly, or whatever other application you're using. So Plover would never be able to 100% reliably know what you want.  I think it's better in the long run to get used to keeping track of what state Plover is in and adjusting as necessary.

You can also do things like defining your strokes that might switch form fields or applications so that they end with the attachment operator `{^}` to avoid an unnecessary space without using an extra stroke.

**Reply by triallax** [reply]
I understand, thanks again.
